packages/discord.js/src/managers/GuildMemberManager.js

```diff
@@ -441,6 +441,32 @@ class GuildMemberManager extends CachedManager {
     return this.guild.bans.remove(user, reason);
   }
 
+  /**
+   * Adds a role to a member.
+   * @param {GuildMemberResolvable} user The member to add a role to
+   * @param {RoleResolvable} role The role to add
+   * @param {string} [reason] The reason for adding the role
+   * @returns {Promise<void>}
+   */
+  async addRole(user, role, reason) {
```

Comment:
```suggestion
  async addRole({ user, role, reason }) {
```

We want this to take an object, also refactor docs/typings appropriately.

Reply:
We made the user a separate argument, for consistency. And above, we have a conversation about the why the reason and role shouldn't be in the same object

Reply:
> We made the user a separate argument, for consistency. And above, we have a conversation about the why the reason and role shouldn't be in the same object

I left this comment after asking @iCrawl if we should be making it an object - the reason I suggested it in the first place is that its consistent with what we're going for across the whole library.

Reply:
That PR moved the name and reason parameters to the object in situations where you’re most often passing an object already. I don’t think it makes sense here because you’re only adding a role and nothing else, with the ability to provide a reason if you want

Reply:
No. If you had actually looked through it, you would have also noticed:
> GuildEmojiManager#create now takes attachment, and name as a part of options

> GuildStickerManager#create now takes file, name, and tags as a part of options

Outside of a few cases (namely `setX(x, reason)` and some other similar patterns), we've decided that these sorts of methods should generally just be taking objects. My PR could only cover so much, so this is not perfectly consistent across the library at the moment, but it's what we're heading towards.

Reply:
This is similar to those setX methods. There’s only 2 parameters and I believe all the ones you listen require or allow more than 2 parameters to be passed